pkg/infrastructure/ibmcloud/clusterapi/clusterapi.go

```diff
@@ -24,5 +29,86 @@ func (Provider) BootstrapHasPublicIP() bool { return false }
+	// Before Provisioning IBM Cloud Infrastructure for the Cluster, we must perform the following.
+	// 1. Create the Resource Group to house cluster resources, if necessary (BYO RG).
+	// 2. Create a COS Instance and Bucket to host the RHCOS Custom Image file.
+	// 3. Upload the RHCOS image to the COS Bucket.
+	// 4. Add IAM Authorization for VPC Image Service to access the COS Object/Bucket/Instance.
+
+	// Setup IBM Cloud Client.
+	metadata := ibmcloudic.NewMetadata(in.InstallConfig.Config)
+	client, err := metadata.Client()
+	if err != nil {
+		return fmt.Errorf("failed creating IBM Cloud client: %w", err)
+	}
+	region := in.InstallConfig.Config.Platform.IBMCloud.Region
+
+	// Create cluster's Resource Group, if necessary (BYO RG is supported).
+	resourceGroupName := in.InfraID
+	if in.InstallConfig.Config.Platform.IBMCloud.ResourceGroupName != "" {
+		resourceGroupName = in.InstallConfig.Config.Platform.IBMCloud.ResourceGroupName
+	}
+
+	// Check whether the Resource Group already exists.
+	resourceGroup, err := client.GetResourceGroup(ctx, resourceGroupName)
+	if err != nil {
+		// If Resource Group cannot be found, but it was provided in install-config (use existing RG), raise an error.
+		// We could create the Resource Group, defined by user, but that might make resource cleanup more difficult.
+		if in.InstallConfig.Config.Platform.IBMCloud.ResourceGroupName != "" {
+			return fmt.Errorf("provided resource group not found: %w", err)
+		}
+	}
+
+	// Create Resource Group if it wasn't found (and isn't expected to be an existing RG).
+	if resourceGroup == nil {
+		if err := client.CreateResourceGroup(ctx, resourceGroupName); err != nil {
+			return fmt.Errorf("failed creating new resource group: %w", err)
+		}
+		// Retrieve the newly created resource group
+		resourceGroup, err = client.GetResourceGroup(ctx, resourceGroupName)
+		if err != nil {
+			return fmt.Errorf("failed retrieving new resource group: %w", err)
+		}
+	}
+
+	// Create a COS Instance and Bucket to host the RHCOS image file.
+	// NOTE(cjschaef): Support to use an existing COS Object (RHCO image file) or VPC Custom Image could be added to skip this step.
+	cosInstanceName := fmt.Sprintf("%s-cos", in.InfraID)
+	cosInstance, err := client.GetCOSInstanceByName(ctx, cosInstanceName)
+	if err != nil {
+		cosInstance, err = client.CreateCOSInstance(ctx, cosInstanceName, *resourceGroup.ID)
+		if err != nil {
+			return fmt.Errorf("failed creating RHCOS image COS instance: %w", err)
+		}
+	}
+	bucketName := fmt.Sprintf("%s-vsi-image", in.InfraID)
+	_, err = client.GetCOSBucketByName(ctx, *cosInstance.ID, bucketName, region)
+	if err != nil {
+		err = client.CreateCOSBucket(ctx, *cosInstance.ID, bucketName, region)
+		if err != nil {
+			return fmt.Errorf("failed creating RHCOS image COS bucket: %w", err)
+		}
+	}
+
+	// Upload the RHCOS image to the COS Bucket.
+	cachedImage, err := cache.DownloadImageFile(in.RhcosImage.Name(), cache.InstallerApplicationName)
```

Comment:
```suggestion
	cachedImage, err := cache.DownloadImageFile(in.RhcosImage.ControlPlane, cache.InstallerApplicationName)
```
The `Name()` of the asset is just "Image" which is not what you want here.

Reply:
updated per other comment.